projects/plugins/jetpack/_inc/client/search/dashboard/index.jsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import React, { Fragment, useMemo } from 'react';
+import { connect } from 'react-redux';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import restApi from 'rest-api';
+import Masthead from 'components/masthead';
+import LoadingPlaceHolder from 'components/loading-placeholder';
+import ModuleControl from './module-control';
+import MockedInstantSearch from './mocked-instant-search';
+import { getPlanClass } from 'lib/plans/constants';
+import './style.scss';
+
+/**
+ * State dependencies
+ */
+import {
+	getSitePlan,
+	hasActiveSearchPurchase as selectHasActiveSearchPurchase,
+	isFetchingSitePurchases,
+} from 'state/site';
+import { setInitialState, getApiNonce, getApiRootUrl } from 'state/initial-state';
+
+const useComponentWillMount = func => {
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	useMemo( func, [] );
+};
+
+/**
+ * SearchDashboard component definition.
+ *
+ * @param {object} props - Component properties.
+ * @returns {React.Component} Search dashboard component.
+ */
+function SearchDashboard( props ) {
+	// NOTE: API root and nonce must be set before any components are mounted!
+	const { apiRootUrl, apiNonce, setInitialState: dispatchedSetInitialState } = props;
+
+	useComponentWillMount( () => {
+		apiRootUrl && restApi.setApiRoot( apiRootUrl );
+		apiNonce && restApi.setApiNonce( apiNonce );
+		dispatchedSetInitialState && dispatchedSetInitialState();
```

Comment:
Nit: I've not seen this pattern of prefixing the variable with `dispatched`; is this a common pattern elsewhere? Otherwise, I'd recommend removing it.

Reply:
The piece of code is from the performance dashboard. I guess they just wanted a different name as the selector has `setInitialState` as its name. Renamed to `setSearchDashboardInitialState` in fe9106924